scripts/exgfs_wave_nawips.sh

```diff
@@ -71,29 +71,19 @@ while [ ${fhcnt} -le ${FHMAX_WAV} ]; do
                grdIDout= ;;
     esac
     GRIBIN="${COM_WAVE_GRID}/${RUNwave}.${cycle}.${grdIDin}.f${fhr}.grib2"
-    GRIBIN_chk=${GRIBIN}.idx
-
-    icnt=1
-    while [ ${icnt} -lt 1000 ]; do
-      if [ -r ${GRIBIN_chk} ] ; then
-        break
-      else
-        let "icnt=icnt+1"
-        sleep 20
-      fi
-      if [ ${icnt} -ge ${maxtries} ]; then
-        msg="ABORTING after 5 minutes of waiting for ${GRIBIN}."
-        echo ' '
-        echo '**************************** '
-        echo '*** ERROR : NO GRIB FILE *** '
-        echo '**************************** '
-        echo ' '
-        echo ${msg}
-        set_trace
-        echo "${RUNwave} ${grdID} ${fhr} prdgen ${date} ${cycle} : GRIB file missing." >> ${wavelog}
-        err=1;export err;${errchk} || exit ${err}
-      fi
-    done
+    sleep_interval=20
+    if ! wait_for_file "${GRIBIN}" "${sleep_interval}" "${maxtries}"; then
```

Comment:
It's important to key off of the index file so processing doesn't start prematurely.
```suggestion
    GRIBIN_chk="${GRIBIN}.idx"
    if ! wait_for_file "${GRIBIN_chk}" "${sleep_interval}" "${maxtries}"; then
```